### PR TITLE
add-time-limit

### PR DESCRIPTION
**Pull request by TTezcan**

### CODE REVIEW

File: src/Groensch/NewRelic/CustomEventHandler/AutoBulkHttp.php [hunk 1]
```diff
@@ -50,8 +60,8 @@ class AutoBulkHttp implements CustomEventHandlerInterface
     public function __construct(HttpInsertApi $httpApi)
     {
         $this
-            ->setNewRelicHttpApi($httpApi)
-        ;
+            ->setNewRelicHttpApi($httpApi);
+        $this->startTimer();
```
Comment (roenschg) [inline src/Groensch/NewRelic/CustomEventHandler/AutoBulkHttp.php]: Please format like this:

```
$this
    ->setNewRelicHttpApi($httpApi)
    ->startTime()
;
```

File: tests/Groensch/NewRelic/CustomEventHandler/AutoBulkHttpTest.php [hunk 2]
```diff
@@ -169,4 +169,99 @@ public function testAutoFlushBufferIfSizeLimitIsReached()
...
+            ->getMockBuilder(AutoBulkHttp::class)
+            ->setConstructorArgs([$httpInsertApiMock])
+            ->setMethods()
+            ->getMock();
+        $autoBulkMock->setTimeToPassInSec(3);
+
+        for ($i = 0; $i < 3; $i++) {
+            $autoBulkMock->recordCustomEvent('test', $data);
+        }
+        sleep(1);
```
Comment (roenschg) [inline tests/Groensch/NewRelic/CustomEventHandler/AutoBulkHttpTest.php]: Sleep slows down the tests. I would recommend to test the actual isTimeOver() method first and the mock it in every test by returning the true or false.

File: src/Groensch/NewRelic/CustomEventHandler/AutoBulkHttp.php [hunk 3]
```diff
@@ -37,6 +37,16 @@ class AutoBulkHttp implements CustomEventHandlerInterface
     private $customEventBuffer = "";
     private $customEventBufferCount = 0;
 
+    /**
+     * @var int
+     */
+    protected $timeToPassInSec = 30;
```
Comment (roenschg) [inline src/Groensch/NewRelic/CustomEventHandler/AutoBulkHttp.php:44]: Please consider to make this as a configurable optional parameter in the constructor.

File: src/Groensch/NewRelic/CustomEventHandler/AutoBulkHttp.php [hunk 4]
```diff
@@ -37,6 +37,16 @@ class AutoBulkHttp implements CustomEventHandlerInterface
...
+
+    /**
+     * @var int
+     */
+    protected $lastTimeInEpoch;
```
Comment (roenschg) [inline src/Groensch/NewRelic/CustomEventHandler/AutoBulkHttp.php]: Please consider a different naming like: lastTimeBufferWasFlushed()

**Review by roenschg — CHANGES_REQUESTED** [review]
Please fix the following points first:
- See the comments at the code
- Check with vendor/bin/phpcs for the code style
- Please also create a PR for the 7.x version as well with your changes.

File: src/Groensch/NewRelic/InvalidArgumentException.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * SOFTWARE.
+ */
+
+
+namespace Groensch\NewRelic;
+
+/**
+ * Class InvalidArgumentException
+ */
+class INVALIDARGUMENTEventException extends \Exception
```
Comment (roenschg) [inline src/Groensch/NewRelic/InvalidArgumentException.php]: What happened here to the name of the class? And where is it used?

**Review by roenschg — CHANGES_REQUESTED** [review]

**Review by roenschg — APPROVED** [review]

**Comment by andreaskweber** [comment]
:D :D :D :D

**Pull request merged**